app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/Indexer/ModifySelectInProductPriceIndexFilter.php

```diff
@@ -84,7 +94,7 @@ public function aroundModifyPrice(
             $select->from(['price_index' => $priceTable->getTableName()], []);
             $priceEntityField = $priceTable->getEntityField();
 
-            if ($this->resourceConnection->getConnection()->isTableExists($stockTable)) {
+            if (!$this->isDefaultStock($stock) && $this->resourceConnection->getConnection()->isTableExists($stockTable)) {
                 $select->joinInner(
```

Comment:
Looks like this join should remain regardless of stock

Reply:
@ishakhsuvarov This join was done for getting SKU from catalog_product_entity and using it for joining inventory_stock_% table because it doesn't have product_id.
We don't need sku if we use cataloginventory_stock_status because it has product_id
Do you see any other reason?